core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2700,11 +2727,12 @@ private void setUfsAcl(LockedInodePath inodePath)
             + "UFS: " + ufsUri + ". This has no effect on the underlying object.");
       } else {
         try {
-          ufs.setAcl(ufsUri, inode.getACL());
+          List<AclEntry> entries = new ArrayList<>(inode.getACL().getEntries());
```

Comment:
is `inode.getACL().getEntries()` expensive?

Reply:
It is not expensive. This method should be dominated by the call to ufs to update ACLs.